### PR TITLE
feat(runtime)!: migrate from Node.js to Bun with standalone executable support

### LINKED ISSUE #249: Migrate runtime from Node.js to Bun with standalone executable distribution
## Context

The current package is a Node.js/npm source distribution. Two repo-specific blockers make the existing Bun migration plan incomplete:

- `src/core/file-io.js` and its tests depend on `better-sqlite3`, which cannot be bundled into a Bun standalone executable.
- Multiple runtime paths spawn package-internal `.js` files by filesystem path (`src/cli/index.js`, `src/core/orchestrator.js`, `src/ui/endpoints/job-control-endpoints.js`). That approach does not map cleanly to a compiled single-file binary.
- Production UI serving currently depends on `express.static()` reading `src/ui/dist` from the filesystem. A compiled binary must embed those assets explicitly.
- The core product behavior depends on loading pipeline task modules from the consumer's filesystem at runtime. That behavior must keep working after compilation.

## Goal

Run the project under Bun and produce a standalone `pipeline-orchestrator` binary with `bun build --compile`. The binary must support `init`, `submit`, `start`, orchestrator-driven job execution, task restart/start endpoints, and loading task modules from an external pipeline root. Node.js runtime compatibility is not part of this issue.

## Non-Goals

- Rewriting the HTTP server to `Bun.serve()`
- Cross-platform release automation or GitHub release asset publishing
- Adding a compatibility adapter that preserves `better-sqlite3`
- Repo-wide `import.meta` rewrites unrelated to internal asset resolution or internal child-process entrypoints

## Acceptance Criteria

1. Source and tests no longer import `better-sqlite3`; runtime SQLite usage is `bun:sqlite` only.
2. No runtime code spawns package-internal `.js` files by path. Internal child processes re-exec the CLI entrypoint with hidden subcommands.
3. `pipeline-orchestrator start` works from source under Bun and from the compiled binary.
4. Job execution started by the orchestrator and by the job-control endpoints works from source under Bun and from the compiled binary.
5. Production UI assets are served from embedded files in the compiled binary, and the source package still serves the existing `src/ui/dist` build from disk.
6. External pipeline task modules referenced from the consumer pipeline root still load at runtime when invoked from the compiled binary.
7. `package.json` runtime metadata and the repo lockfile reflect Bun as the supported runtime.
8. Focused Bun tests covering SQLite, self-reexec command wiring, embedded UI asset serving, and compiled-binary smoke behavior pass.

## Implementation Steps

1. Update runtime metadata in `package.json` for Bun-only development and execution. Replace the `node` engine with a `bun` engine, change scripts that execute local source files from `node ...` to `bun ...`, remove the `better-sqlite3` dependency, replace the npm lockfile with a Bun lockfile, and keep the published package contents unchanged except for the runtime metadata required by this migration.
2. Replace `better-sqlite3` with direct `bun:sqlite` usage in `src/core/file-io.js`. Import `Database` from `bun:sqlite`, keep `getDB(options)` synchronous, change WAL setup to `db.exec("PRAGMA journal_mode = WAL;")`, and update the JSDoc types and error text that currently name `better-sqlite3`.
3. Update SQLite tests to the Bun API used by this repo. Change `tests/batch-runner.test.js` and `tests/file-io-sqlite.test.js` to import `Database` from `bun:sqlite`, keep `prepare(...).run/get/all`, keep the readonly option assertion, and query `PRAGMA journal_mode` with SQL instead of using a `db.pragma()` helper.
4. Extract the pipeline-runner script body in `src/core/pipeline-runner.js` into an exported function that accepts `jobId` and reads the same environment variables as today. Leave a thin direct-execution path in the module so source-mode invocation still works during the migration.
5. Add hidden internal subcommands to `src/cli/index.js` for starting the UI server, starting the orchestrator, and running a single pipeline job. Each hidden command must call the existing runtime entrypoint directly: `startServer(...)`, `startOrchestrator(...)`, and the new exported pipeline-runner function.
6. Add a shared helper that builds self-reexec spawn arguments for both source mode and compiled-binary mode. In source mode, the helper must include the CLI source file path before the hidden command so `bun` re-runs `src/cli/index.js`. In compiled-binary mode, the helper must pass only the hidden command and its arguments because the entrypoint is already embedded in `process.execPath`.
7. Replace the `start` command child-process spawns in `src/cli/index.js` to use the self-reexec helper and hidden commands instead of `src/ui/server.js` and `src/cli/run-orchestrator.js` file paths. Preserve the current environment variables, output piping, and kill-others-on-fail behavior.
8. Replace pipeline-runner spawns in `src/core/orchestrator.js` and `src/ui/endpoints/job-control-endpoints.js` to use the same self-reexec helper and hidden pipeline-runner command instead of `../../core/pipeline-runner.js`. Keep the current environment contract (`PO_ROOT`, `PO_DATA_DIR`, `PO_START_FROM_TASK`, `PO_RUN_SINGLE_TASK`, and related variables) unchanged.
9. Remove `src/cli/run-orchestrator.js` only after all callers have been moved to the hidden CLI command path and the remaining tests no longer depend on the file-path entrypoint. If keeping the file is simpler, reduce it to a thin wrapper that delegates to the same orchestrator start function used by the hidden command.
10. Generate an explicit embedded UI asset map from `src/ui/dist` as part of the UI build. Add a build-time script that walks the built dist directory and writes a generated module with one static `with { type: "file" }` import per asset plus a request-path-to-embedded-file map. The generated module must include `index.html` and every file under `src/ui/dist/assets`.
11. Update production asset serving in `src/ui/express-app.js` to serve from the generated embedded asset map when embedded files are available and to fall back to the existing filesystem `src/ui/dist` paths when running from source. Do not rely on `express.static()` alone for compiled binaries; add explicit handlers that resolve the embedded file path and send the correct asset content.
12. Limit path-resolution changes to files that resolve package-internal runtime assets or child entrypoints. Update `src/cli/index.js`, `src/core/orchestrator.js`, `src/ui/endpoints/job-control-endpoints.js`, and `src/ui/express-app.js` as needed, and do not perform a repo-wide `import.meta.url` rewrite.
13. Add focused tests for the new runtime boundaries. Cover direct `bun:sqlite` usage, the self-reexec argument builder in source and compiled modes, CLI `start` spawn wiring, job-control spawn wiring, and production asset serving from the generated embedded asset map.
14. Add a compiled-binary smoke test that builds `src/cli/index.js` with `bun build --compile`, runs the resulting executable against a temporary pipeline root, verifies `init` and `submit`, boots `start`, and executes a pipeline whose task registry lives outside the binary on the filesystem. The test must fail if runtime task-module loading, internal child-process spawning, or embedded UI asset serving is broken.
15. Update the Bun-facing automation required by this issue to stay green. Change the existing CI/runtime command path to install dependencies with Bun and run only the focused migration tests plus the compiled-binary smoke test added in the previous steps.

### PR DESCRIPTION
**Pull request by ryan-mahoney**
# Why

The current package is a Node.js/npm source distribution with several architectural constraints blocking standalone binary compilation:

- `src/core/file-io.js` depends on `better-sqlite3`, which cannot be bundled into a Bun standalone executable
- Multiple runtime paths spawn package-internal `.js` files by filesystem path, which doesn't map cleanly to a compiled single-file binary
- Production UI serving depends on `express.static()` reading from filesystem, but compiled binaries must embed assets explicitly
- External pipeline task modules must continue loading from consumer's filesystem at runtime

# What Changed

- **Runtime migration**: Replaced Node.js with Bun as the supported runtime (engine requirement: `bun >=1.1.0`)
- **SQLite migration**: Replaced `better-sqlite3` with `bun:sqlite` for embedded database support
- **Self-reexec pattern**: Added hidden CLI subcommands for internal child processes (UI server, orchestrator, pipeline-runner) instead of spawning `.js` files by path
- **Embedded UI assets**: Added build-time script (`scripts/generate-embedded-assets.js`) to generate embedded asset map for compiled binary distribution
- **Asset serving**: Updated `src/ui/express-app.js` to serve from embedded assets when available, fallback to filesystem for source mode
- **Lockfile migration**: Replaced `package-lock.json` with `bun.lock`
- **Test coverage**: Added focused Bun tests for SQLite, self-reexec command wiring, embedded UI asset serving, and compiled-binary smoke behavior

## Key Files Changed

| File | Purpose |
|------|---------|
| `package.json` | Updated all scripts to use `bun`, changed engine requirement |
| `src/core/file-io.js` | Migrated to `bun:sqlite` |
| `src/cli/index.js` | Added hidden subcommands for internal entrypoints |
| `src/cli/self-reexec.js` | New helper for source/compiled spawn argument building |
| `src/core/pipeline-runner.js` | Extracted function for CLI command invocation |
| `src/ui/express-app.js` | Added embedded asset serving support |
| `scripts/generate-embedded-assets.js` | New build-time script for asset embedding |
| `tests/bun-migration.bun.test.js` | New tests for runtime boundaries |
| `tests/compiled-binary.bun.test.js` | New smoke test for compiled binary |

# How Was This Tested

- Bun-specific unit tests for SQLite, self-reexec, and embedded assets
- Compiled binary smoke test that builds executable and verifies `init`, `submit`, `start`, and pipeline execution
- Existing test suite passes under Bun runtime
- Manual verification of demo workflow from source and compiled binary

# Risks & Rollback

- **Risk**: Large binary file (`cli` at 72MB) may cause repo bloat. Consider Git LFS or excluding from PR in future iterations.
- **Rollback**: Revert this commit and restore Node.js runtime from previous commit history.

# Performance / Security / Accessibility

- Performance: Bun runtime provides faster startup and native SQLite performance
- Security: No new security concerns introduced
- Accessibility: No UI changes in this migration

# Linked Issues

Closes #249

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,20 +21,21 @@
...
-    "demo:orchestrator": "PO_ROOT=demo NODE_ENV=production node -e \"import('./src/core/orchestrator.js').then(m => m.startOrchestrator({ dataDir: process.env.PO_ROOT || 'demo' })).catch(err => { console.error(err); process.exit(1) })\"",
-    "demo:all": "NODE_ENV=development PO_ROOT=demo npm run ui:build && concurrently \"npm:demo:ui\" \"npm:demo:orchestrator\" --kill-others-on-fail",
-    "demo:prod": "npm run ui:build && NODE_ENV=production PO_ROOT=demo node src/ui/server.js"
+    "prepack": "bun run ui:build",
+    "ui:prod": "bun src/ui/server.js",
+    "demo:ui": "NODE_ENV=production PO_ROOT=demo bun src/ui/server.js",
+    "demo:orchestrator": "PO_ROOT=demo NODE_ENV=production bun -e \"import('./src/core/orchestrator.js').then(m => m.startOrchestrator({ dataDir: process.env.PO_ROOT || 'demo' })).catch(err => { console.error(err); process.exit(1) })\"",
+    "demo:all": "NODE_ENV=development PO_ROOT=demo bun run ui:build && concurrently \"bun:demo:ui\" \"bun:demo:orchestrator\" --kill-others-on-fail",
+    "demo:prod": "bun run ui:build && NODE_ENV=production PO_ROOT=demo bun src/ui/server.js",
+    "generate:embedded-assets": "bun scripts/generate-embedded-assets.js"
```
[the remark this answers is not in the document]
Comment (ryan-mahoney) [reply package.json]: Fixed. `ui:build` now runs `vite build && bun run generate:embedded-assets`, so `prepack` inherits the asset-map regeneration and keeps `src/ui/embedded-assets.js` in sync with the latest hashed `src/ui/dist` output.
